### PR TITLE
Excluding false positives in route filtering in Generator@getRoutes

### PR DESCRIPTION
**Pull request by Carnicero90**
Str::startsWith($route->uri, config('scramble.api_path', 'api') may actually includes web routes (I.E.: '/apicoltore').
Kinda preventing that, plus making config('scramble.api_path') a little more error forgiving (by trimming leading and trailing slashes before matching it with route URIs).

### CODE REVIEW

File: src/Generator.php [hunk 1]
```diff
@@ -141,7 +141,10 @@ private function getRoutes(): Collection
                 $routeResolver = Scramble::$routeResolver ?? function (Route $route) {
                     $expectedDomain = config('scramble.api_domain');
 
-                    return Str::startsWith($route->uri, config('scramble.api_path', 'api'))
+                    $routePrefix = trim(config('scramble.api_path', 'api'), '/');
+                    $routeRegex = '/^' . preg_quote($routePrefix, '/') . '((\/.+)|$)/';
+
+                    return preg_match($routeRegex, $route->uri)
```
Comment (romalytvynenko) [inline src/Generator.php:147]: Does this regexp equal to this approach? If so, I'd prefer `startsWith`.

```suggestion
                    return Str::startsWith($route->uri, rtrim(config('scramble.api_path', 'api'), '/').'/');
```

File: tests/Generator/Routes/RoutesFilteringTest.php
Comment (romalytvynenko) [inline tests/Generator/Routes/RoutesFilteringTest.php:1]: Formatting fixes needed (line 12 is redundant empty line, broken formatting on lines 77-79).

Also, using data providers can simplify this test file to just a single test with few data items.

**Review by romalytvynenko — COMMENTED** [review]
Thanks for the PR. Few questions and suggestions before I merge.

File: tests/Generator/Routes/RoutesFilteringTest.php
Comment (Carnicero90) [reply tests/Generator/Routes/RoutesFilteringTest.php:1]: my bad, forgot to run pint I guess.

File: src/Generator.php (hunk 1 above)
Comment (Carnicero90) [reply src/Generator.php:147]: I think it should be `Str::startsWith($route->uri.'/', rtrim(config('scramble.api_path', 'api'), '/').'/')`: with that fix yeah, it actually does the very same thing. Gonna change that.

**Comment by romalytvynenko** [comment]
Closing due to inactivity

**Comment by Carnicero90** [comment]
Oh my apologies (and merry Christmas), I completely forgot about that pr: I'll update my branch according to your suggestions and resubmit it

**Pull request closed without merging**